Apps/WebClient/src/Server/Services/UserProfileService.cs

```diff
@@ -125,6 +125,7 @@ public RequestResult<UserProfileModel> GetUserProfile(string hdid, DateTime jwtA
             RequestResult<TermsOfServiceModel> termsOfServiceResult = this.GetActiveTermsOfService();
 
             UserProfileModel userProfile = UserProfileModel.CreateFromDbModel(retVal.Payload);
+            userProfile.IdentityProviderEmail = jwtEmailAddress;
```

Comment:
why is this necessary? Wouldnt both be the same?

Reply:
This is where IdentityProviderEmail is populated.  `jwtEmailAddress` is fetched in GetUserProfile() in the controller.

Reply:
AFAIK the JWT is read on the backend to make sure that is validated, all other flows work the same